packaging_automation/test_citus_package.py

```diff
@@ -0,0 +1,108 @@
+    undefined = {"name": "undefined", "docker_image_name": "undefined"}
+
+
+def parse_os_release(os_release: str) -> TestPlatform:
+    result = TestPlatform.undefined
+    for tp in TestPlatform:
+        if tp.value["name"] == os_release:
+            result = tp
+    return result
+
+
+def get_postgres_versions_from_matrix_file(project_version: str):
+    r = requests.get(POSTGRES_MATRIX_WEB_ADDRESS, allow_redirects=True)
+
+    with open(POSTGRES_MATRIX_FILE, 'wb') as writer:
+        writer.write(r.content)
+    pg_versions = get_supported_postgres_release_versions(POSTGRES_MATRIX_FILE, project_version)
+    # Since string expressions with double quotes cause problems in pipeline, string is converted into int
+    # pg_versions = [int(pg_ver) for pg_ver in pg_versions]
+
+    return pg_versions
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--prj_ver', required=True)
+    parser.add_argument('--pg_major_version')
+    parser.add_argument("--os_release", choices=[t.value["name"] for t in TestPlatform])
+
+    args = parser.parse_args()
+    test_platform = parse_os_release(args.os_release)
+    minor_prj_ver = get_minor_version(args.prj_ver)
+
+    platform = args.os_release
+
+    postgres_versions = get_postgres_versions_from_matrix_file(args.prj_ver)
+
+    print("Pg Versions: ")
+    print(postgres_versions)
+
+    os.chdir("test-images")
+    return_codes = {}
+
+    if args.pg_major_version:
+        postgres_versions = [p for p in postgres_versions if p == args.pg_major_version]
+
+    print(postgres_versions)
+    if len(postgres_versions) == 0:
+        raise ValueError("At least one supported postgres version is required")
+
+    for postgres_version in postgres_versions:
+        docker_image_name = f"test:{test_platform.value['docker_image_name']}-{postgres_version}"
+        build_command = (f"docker build -t {docker_image_name} "
+                         f"-f {test_platform.value['docker_image_name']}/Dockerfile "
+                         f"--build-arg CITUS_VERSION={args.prj_ver} --build-arg PG_MAJOR={postgres_version}   "
+                         f"--build-arg CITUS_MAJOR_VERSION={minor_prj_ver} .")
+        print(build_command)
+        return_build = run_command(build_command)
+        return_run = run_command(
+            f"docker run  -e POSTGRES_VERSION={postgres_version} {docker_image_name} ")
+        return_codes[f"{docker_image_name}-build"] = return_build
+        return_codes[f"{docker_image_name}-run"] = return_run
+
+    error_exists = False
+    print("-----------------Summary Report------------------")
+    for key, value in return_codes.items():
+        if value > 0:
+            error_exists = True
+        print(f"{key}: {'Success' if value == 0 else f'Fail. ErrorCode: {value}'}")
+    summary_error = 'FAILED :(' if error_exists else 'SUCCESS :)'
+    print(f'{summary_error}-------------------------------------------------')
```

Comment:
very minor, for cosmetic purposes :)
```suggestion
    print(f'------------------------{summary_error}'------------------------')
```

(btw, I really liked the "SUCCESS :)" message here, it looks so positive :D)